lib/bundler/definition.rb

```diff
@@ -644,7 +644,8 @@ def converge_sources
       if !locked_gem_sources.empty? && !actual_remotes.empty?
         locked_gem_sources.each do |locked_gem|
           # Merge the remotes from the Gemfile into the Gemfile.lock
-          changes |= locked_gem.replace_remotes(actual_remotes)
+          changes |= !locked_gem.equivalent_remotes?(actual_remotes)
```

Comment:
My first approach was to do it there, but I'm not sure if it would have any other implications, for one it would mean that that the hosts are not actually replaced, so the host with the username and password would not be used. Maybe thats not important but it seemed pretty invasive.
If you think its ok to do that there I'm happy to move it